cardano-cli/src/Cardano/CLI/Run/Debug/TransactionView.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Cardano.CLI.Run.Debug.TransactionView
+  ( runTransactionViewCmd
+  )
+where
+
+import           Cardano.Api
+
+import           Cardano.CLI.Commands.Debug.TransactionView
+import           Cardano.CLI.Json.Friendly (FriendlyFormat (..), friendlyTx, friendlyTxBody)
+import           Cardano.CLI.Read
+import           Cardano.CLI.Types.Common
+import           Cardano.CLI.Types.Errors.TxCmdError
+
+import           Data.Function ((&))
+
+runTransactionViewCmd
+  :: ()
+  => TransactionViewCmdArgs
+  -> ExceptT TxCmdError IO ()
+runTransactionViewCmd
+  TransactionViewCmdArgs
+    { outputFormat
+    , mOutFile
+    , inputTxBodyOrTxFile
+    } =
+    case inputTxBodyOrTxFile of
+      InputTxBodyFile (File txbodyFilePath) -> do
+        txbodyFile <- liftIO $ fileOrPipe txbodyFilePath
+        unwitnessed <-
+          firstExceptT TxCmdTextEnvCddlError . newExceptT $
+            readFileTxBody txbodyFile
+        InAnyShelleyBasedEra era txbody <- pure $ unIncompleteCddlTxBody unwitnessed
+        -- Why are we differentiating between a transaction body and a transaction?
+        -- In the case of a transaction body, we /could/ simply call @makeSignedTransaction []@
+        -- to get a transaction which would allow us to reuse friendlyTxBS. However,
```

Comment:
This is something the new api would fix because by default we would either produce an unsigned tx or a signed tx. No need to think about transaction bodies that are missing witnesses.